Jobs/v4 API surface appears to have changed

In PR #2384 we upgraded the BOM and got in both Java 8 & Java 11 testing:
```
------------------------------------------------------------
- testing jobs/v4
------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateCompany.java:[25,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateJob.java:[25,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/JobSearchAutoCompleteJobTitle.java:[25,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/JobSearchListJobs.java:[25,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/JobSearchListCompanies.java:[25,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/CustomRankingSearchJobs.java:[28,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/JobSearchDeleteJob.java:[24,38] error: cannot find symbol
  symbol:   class JobWithTenantName
  location: package com.google.cloud.talent.v4beta1
[ERROR] /tmpfs/src/github/java-docs-samples/jobs/v4/src/main/java/com/example/jobs/CommuteSearchJobs.java:[30,38] error: cannot find symbol
  symbol:   class TenantOrProjectName
```

--- a/jobs/v4/src/main/java/com/example/jobs/CommuteSearchJobs.java
+++ b/jobs/v4/src/main/java/com/example/jobs/CommuteSearchJobs.java
@@ -27,7 +27,6 @@ import com.google.cloud.talent.v4beta1.RequestMetadata;
 import com.google.cloud.talent.v4beta1.SearchJobsRequest;
 import com.google.cloud.talent.v4beta1.SearchJobsResponse;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import com.google.protobuf.Duration;
 import com.google.type.LatLng;
 import java.io.IOException;
@@ -44,7 +43,7 @@ public class CommuteSearchJobs {
   // Search Jobs with histogram queries.
   public static void searchJobs(String projectId, String tenantId) throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
       String domain = "www.example.com";
       String sessionId = "Hashed session identifier";
       String userId = "Hashed user identifier";

--- a/jobs/v4/src/main/java/com/example/jobs/CustomRankingSearchJobs.java
+++ b/jobs/v4/src/main/java/com/example/jobs/CustomRankingSearchJobs.java
@@ -18,14 +18,12 @@ package com.example.jobs;
 
 // [START job_search_custom_ranking_search]
 
-import com.google.cloud.talent.v4beta1.HistogramQuery;
 import com.google.cloud.talent.v4beta1.Job;
 import com.google.cloud.talent.v4beta1.JobServiceClient;
 import com.google.cloud.talent.v4beta1.RequestMetadata;
 import com.google.cloud.talent.v4beta1.SearchJobsRequest;
 import com.google.cloud.talent.v4beta1.SearchJobsResponse;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 
 public class CustomRankingSearchJobs {
@@ -40,7 +38,7 @@ public class CustomRankingSearchJobs {
   // Search Jobs using custom rankings.
   public static void searchCustomRankingJobs(String projectId, String tenantId) throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
       String domain = "www.example.com";
       String sessionId = "Hashed session identifier";
       String userId = "Hashed user identifier";

--- a/jobs/v4/src/main/java/com/example/jobs/HistogramSearchJobs.java
+++ b/jobs/v4/src/main/java/com/example/jobs/HistogramSearchJobs.java
@@ -25,7 +25,6 @@ import com.google.cloud.talent.v4beta1.RequestMetadata;
 import com.google.cloud.talent.v4beta1.SearchJobsRequest;
 import com.google.cloud.talent.v4beta1.SearchJobsResponse;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 
 public class HistogramSearchJobs {
@@ -42,7 +41,7 @@ public class HistogramSearchJobs {
   public static void searchJobs(String projectId, String tenantId, String query)
       throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
 
       String domain = "http://www.jobUrl.com";
       String sessionId = "Hashed session identifier";

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchAutoCompleteJobTitle.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchAutoCompleteJobTitle.java
@@ -22,7 +22,6 @@ import com.google.cloud.talent.v4beta1.CompleteQueryRequest;
 import com.google.cloud.talent.v4beta1.CompleteQueryResponse;
 import com.google.cloud.talent.v4beta1.CompletionClient;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 
 public class JobSearchAutoCompleteJobTitle {
@@ -39,7 +38,7 @@ public class JobSearchAutoCompleteJobTitle {
   public static void completeQuery(String projectId, String tenantId, String query)
       throws IOException {
     try (CompletionClient completionClient = CompletionClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
       CompleteQueryRequest request =
           CompleteQueryRequest.newBuilder()
               .setParent(parent.toString())

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateClientEvent.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateClientEvent.java
@@ -23,7 +23,6 @@ import com.google.cloud.talent.v4beta1.CreateClientEventRequest;
 import com.google.cloud.talent.v4beta1.EventServiceClient;
 import com.google.cloud.talent.v4beta1.JobEvent;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import com.google.protobuf.Timestamp;
 import java.io.IOException;
 import java.util.Arrays;
@@ -44,7 +43,7 @@ public class JobSearchCreateClientEvent {
   public static void createClientEvent(
       String projectId, String tenantId, String requestId, String eventId) throws IOException {
     try (EventServiceClient eventServiceClient = EventServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
 
       // The timestamp of the event as seconds of UTC time since Unix epoch
       // For more information on how to create google.protobuf.Timestamps

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateCompany.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateCompany.java
@@ -22,7 +22,6 @@ import com.google.cloud.talent.v4beta1.Company;
 import com.google.cloud.talent.v4beta1.CompanyServiceClient;
 import com.google.cloud.talent.v4beta1.CreateCompanyRequest;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 
 public class JobSearchCreateCompany {
@@ -40,7 +39,7 @@ public class JobSearchCreateCompany {
   public static void createCompany(
       String projectId, String tenantId, String displayName, String externalId) throws IOException {
     try (CompanyServiceClient companyServiceClient = CompanyServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
       Company company =
           Company.newBuilder().setDisplayName(displayName).setExternalId(externalId).build();
 

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateJob.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateJob.java
@@ -22,7 +22,6 @@ import com.google.cloud.talent.v4beta1.CreateJobRequest;
 import com.google.cloud.talent.v4beta1.Job;
 import com.google.cloud.talent.v4beta1.JobServiceClient;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
@@ -48,7 +47,7 @@ public class JobSearchCreateJob {
       String jobApplicationUrl)
       throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
       Job.ApplicationInfo applicationInfo =
           Job.ApplicationInfo.newBuilder().addUris(jobApplicationUrl).build();
 

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateJobCustomAttributes.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchCreateJobCustomAttributes.java
@@ -23,10 +23,7 @@ import com.google.cloud.talent.v4beta1.CustomAttribute;
 import com.google.cloud.talent.v4beta1.Job;
 import com.google.cloud.talent.v4beta1.JobServiceClient;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
 
 public class JobSearchCreateJobCustomAttributes {
 
@@ -47,7 +44,7 @@ public class JobSearchCreateJobCustomAttributes {
           String requisitionId)
           throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
 
       // Custom attribute can be string or numeric value, and can be filtered in search queries.
       // https://cloud.google.com/talent-solution/job-search/docs/custom-attributes

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchDeleteCompany.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchDeleteCompany.java
@@ -20,7 +20,6 @@ package com.example.jobs;
 
 import com.google.cloud.talent.v4beta1.CompanyName;
 import com.google.cloud.talent.v4beta1.CompanyServiceClient;
-import com.google.cloud.talent.v4beta1.CompanyWithTenantName;
 import com.google.cloud.talent.v4beta1.DeleteCompanyRequest;
 import java.io.IOException;
 
@@ -38,7 +37,7 @@ public class JobSearchDeleteCompany {
   public static void deleteCompany(String projectId, String tenantId, String companyId)
       throws IOException {
     try (CompanyServiceClient companyServiceClient = CompanyServiceClient.create()) {
-      CompanyName name = CompanyWithTenantName.of(projectId, tenantId, companyId);
+      CompanyName name = CompanyName.ofProjectTenantCompanyName(projectId, tenantId, companyId);
 
       DeleteCompanyRequest request =
           DeleteCompanyRequest.newBuilder().setName(name.toString()).build();

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchDeleteJob.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchDeleteJob.java
@@ -21,7 +21,6 @@ package com.example.jobs;
 import com.google.cloud.talent.v4beta1.DeleteJobRequest;
 import com.google.cloud.talent.v4beta1.JobName;
 import com.google.cloud.talent.v4beta1.JobServiceClient;
-import com.google.cloud.talent.v4beta1.JobWithTenantName;
 import java.io.IOException;
 
 public class JobSearchDeleteJob {
@@ -37,7 +36,7 @@ public class JobSearchDeleteJob {
   // Delete Job.
   public static void deleteJob(String projectId, String tenantId, String jobId) throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      JobName name = JobWithTenantName.of(projectId, tenantId, jobId);
+      JobName name = JobName.ofProjectTenantJobName(projectId, tenantId, jobId);
 
       DeleteJobRequest request = DeleteJobRequest.newBuilder().setName(name.toString()).build();
 

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchGetCompany.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchGetCompany.java
@@ -21,7 +21,6 @@ package com.example.jobs;
 import com.google.cloud.talent.v4beta1.Company;
 import com.google.cloud.talent.v4beta1.CompanyName;
 import com.google.cloud.talent.v4beta1.CompanyServiceClient;
-import com.google.cloud.talent.v4beta1.CompanyWithTenantName;
 import com.google.cloud.talent.v4beta1.GetCompanyRequest;
 import java.io.IOException;
 
@@ -39,7 +38,7 @@ public class JobSearchGetCompany {
   public static void getCompany(String projectId, String tenantId, String companyId)
       throws IOException {
     try (CompanyServiceClient companyServiceClient = CompanyServiceClient.create()) {
-      CompanyName name = CompanyWithTenantName.of(projectId, tenantId, companyId);
+      CompanyName name = CompanyName.ofProjectTenantCompanyName(projectId, tenantId, companyId);
 
       GetCompanyRequest request = GetCompanyRequest.newBuilder().setName(name.toString()).build();
 

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchGetJob.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchGetJob.java
@@ -22,7 +22,6 @@ import com.google.cloud.talent.v4beta1.GetJobRequest;
 import com.google.cloud.talent.v4beta1.Job;
 import com.google.cloud.talent.v4beta1.JobName;
 import com.google.cloud.talent.v4beta1.JobServiceClient;
-import com.google.cloud.talent.v4beta1.JobWithTenantName;
 import java.io.IOException;
 
 public class JobSearchGetJob {
@@ -38,7 +37,7 @@ public class JobSearchGetJob {
   // Get Job.
   public static void getJob(String projectId, String tenantId, String jobId) throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      JobName name = JobWithTenantName.of(projectId, tenantId, jobId);
+      JobName name = JobName.ofProjectTenantJobName(projectId, tenantId, jobId);
 
       GetJobRequest request = GetJobRequest.newBuilder().setName(name.toString()).build();
 

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchListCompanies.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchListCompanies.java
@@ -22,7 +22,6 @@ import com.google.cloud.talent.v4beta1.Company;
 import com.google.cloud.talent.v4beta1.CompanyServiceClient;
 import com.google.cloud.talent.v4beta1.ListCompaniesRequest;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 
 public class JobSearchListCompanies {
@@ -37,7 +36,7 @@ public class JobSearchListCompanies {
   // List Companies.
   public static void listCompanies(String projectId, String tenantId) throws IOException {
     try (CompanyServiceClient companyServiceClient = CompanyServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
 
       ListCompaniesRequest request =
           ListCompaniesRequest.newBuilder().setParent(parent.toString()).build();

--- a/jobs/v4/src/main/java/com/example/jobs/JobSearchListJobs.java
+++ b/jobs/v4/src/main/java/com/example/jobs/JobSearchListJobs.java
@@ -22,7 +22,6 @@ import com.google.cloud.talent.v4beta1.Job;
 import com.google.cloud.talent.v4beta1.JobServiceClient;
 import com.google.cloud.talent.v4beta1.ListJobsRequest;
 import com.google.cloud.talent.v4beta1.TenantName;
-import com.google.cloud.talent.v4beta1.TenantOrProjectName;
 import java.io.IOException;
 
 public class JobSearchListJobs {
@@ -38,7 +37,7 @@ public class JobSearchListJobs {
   // Search Jobs with histogram queries.
   public static void listJobs(String projectId, String tenantId, String filter) throws IOException {
     try (JobServiceClient jobServiceClient = JobServiceClient.create()) {
-      TenantOrProjectName parent = TenantName.of(projectId, tenantId);
+      TenantName parent = TenantName.of(projectId, tenantId);
       ListJobsRequest request =
           ListJobsRequest.newBuilder().setParent(parent.toString()).setFilter(filter).build();
       for (Job responseItem : jobServiceClient.listJobs(request).iterateAll()) {

